073-improve-handling-of-CA-renewals-and-replacements-in-client-based-operands.md

```diff
@@ -0,0 +1,211 @@
+7. The removal of `ca-<timestamp>.crt` from the Secret will cause rolling update of the client-based operand (e.g. Connect).
+   This rolling update will remove the trust to the old CA from the operand as well.
+
+### Examples
+
+The examples for the operands based on Kafka clients will be updated to use the new field:
+```yaml
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+        extension: crt
+```
+
+This will make sure that when used with a Strimzi based cluster, everything - including CA replacement - will work without any outages and issues.
+
+## Affected projects
+
+This proposal affects only the Strimzi Cluster Operator project.
+
+## Backwards compatibility
+
+There is no impact on backwards compatibility.
+This proposal does not remove any Strimzi APIs.
+It only extends them.
+All existing custom resources will work without any change.
+
+## Rejected alternatives
+
+### Using a regular expression instead of extension
+
+One option considered for this proposal was to use a pattern to specify the matching certificates instead of extension.
+However, when using a pattern / regular expression:
+* The correct value would depend on where is the pattern applied (e.g. in Java, in shell when starting the container image, in some different language in the future).
+* For non-expert users it might be hard to understand the exact format and the different regular expression variants.
+* It would be hard to validate the the pattern / regular expression.
+
+That is why this option was rejected.
+
+### Hardcoding the extension
+
+Another considered option was to not add a new field to specify the `extension` at all in the API and just make the `certificate` field optional.
+When the `certificate` field would not be specified, all items from the Secret with the `.crt` extension would be loaded.
+This option was rejected because in some cases, users might use different extension such as `.pem` for public keys and having it hardcoded might make it not work for them.
+
+It was also considered to not add the `extension` field, but instead of hardcoding the `.crt` extension to simply try to load everything from the Secret.
+But in this situation, we might struggle to identify the fields that contain a public key.
+We would likely need to parse them and that might not produce reliable results.
+So this alternative was rejected as well.
+
+### Making `certificate` and `extension` fields completely optional
+
+Another option - similar to the previous one - was to keep the `extension` field.
+But make specifying `extension` (as well as `certicicate`) fully optional.
+For example:
+```yaml
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+```
+
+And when none of them would be specified, it will be treated as if `extension` was set to `crt`.
+This is something what can be considered and added even later if desired.
```

Comment:
This seems reasonable if users are made aware of the default behavior and it aligns with the majority of use cases. But are unintended consequences and risks too severe if overlooked by users?

Reply:
I'm not sure I would be worried so much about the unintended consequences. It is more that doing this means:
* Yet another path in the code
* Yet another thing to test

And I wonder if it really adds something to the user experience (easy of use) or not compared to the user having to add the `extension: crt` to the custom resource.